apps/src/templates/teacherDashboard/MultiSelectGroup.jsx

```diff
@@ -37,10 +40,12 @@ export default function MultiSelectGroup({
             // `checked`.
             required={required ? !Object.values(values).some(v => !!v) : false}
             onCheckedChange={checked => {
-              setValues({
+              const newValues = {
                 ...values,
                 [option.value]: checked
-              });
+              };
+              setValues(newValues);
+              setSelected(Object.keys(newValues).filter(k => newValues[k]));
```

Comment:
ok, I think I have it redone now: https://github.com/code-dot-org/code-dot-org/pull/50248/files#diff-e268ab0ed586a40fbd3f4fe9aa57ebfc25ce9cde740add6a394c01b2db1dec7eR39-R46

the logic is more complicated because I have to remove by value from an array when something is unchecked, but it seems worth it to just store the values once and have them in the format i want.